File: 2 semestre/Laboratório de Desenvolvimento de Algoritmos/Lista de Exercícios 3/src/Ex8.java
//Importando Scanner
import java.util.Scanner;

public class Ex8 {

	public static void main(String[] args){
		
		//Instanciando Scanner
		Scanner input = new Scanner(System.in);
		
		System.out.println("\n\n");
		
		//vars
		double massaI, massaF;
		int h, m ,s, tempo=0;
		
		//Entrada
		System.out.print("Insira a massa inicial do material, em gramas: ");
		massaF = input.nextDouble();
		
		//Processamento
		massaI = massaF;
		
		while( massaF >= 0.5 )
		{
			massaF /= 2;
			tempo += 50;
		}
		
		h = tempo / 3600;
		m = (tempo%3600) / 60;
		s = (tempo%3600) % 60;
		
		//Saída
		System.out.printf("Massa inicial: %.2f\n", massaI);
		System.out.printf("Massa final: %.2f\n", massaF);
		System.out.printf("Tempo: %dh %dm %ds", h, m, s);
		
		
		
	}
}

File: 2 semestre/Técnica de Desenvolvimento de Algoritmos/Listas de Exercicios/Lista 3/ex8.c
#include <stdio.h>

main()
{
    printf("");

    //vars
    float massaF, massaI;
    int totalSegundos=0, h=0, m=0, s=0;

    //Entrada
    printf("Insira a massa inicial do material, em gramas: ");
    scanf("%f", &massaF);

    //Processamento
    massaI = massaF;

    while( massaF > 0.5 )
    {
        massaF /= 2;
        totalSegundos += 50;
    }

    h = totalSegundos / 3600;
    m = (totalSegundos%3600) / 60;
    s = (totalSegundos%3600) % 60;

    printf("\nMassa Inicial: %.2f gramas\n", massaI);
    printf("Massa Final: %.2f gramas\n", massaF);
    printf("Tempo: %ih %dm %is ", h, m, s);

}
